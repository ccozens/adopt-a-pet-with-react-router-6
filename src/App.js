import HomePage from './pages/home';
import SearchPage from './pages/search';
import PetDetailsPage from './pages/detail';
import PetDetailsNotFound from './pages/petDetailsNotFound';
import Navigation from './components/navigation';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';

function App() {
  return (
    <Router>
      <div>
          <Navigation />
        <Routes>
            <Route path="/" element={<HomePage />} />
            <Route path=":type/*" element={<HomePage />} />
            <Route path=":type/:id/*" element={<PetDetailsPage />} />
            <Route path="search/*" element={<SearchPage />} />
            <Route path="pet-details-not-found" element={< PetDetailsNotFound />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
