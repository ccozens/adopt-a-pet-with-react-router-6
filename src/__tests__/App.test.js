import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
// import modules to test
import App from '../App';
import Hero from '../components/hero/index';

afterEach(cleanup);

test('App should look like the snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});

test('hero should render', () => {
  const { getByText } = render(<Hero />);
  expect(getByText('Find your perfect pet'));
});

test('search on home page works', () => {
  render(<App />);
  // assign search bar button
  const button = screen.getByRole('button', { name: 'ðŸ”Ž' });
  // assign search box
  const searchBox = screen.getByRole('textbox', { name: 'search-box' });
  // type in search box
  userEvent.type(searchBox, 'dogs');
  // click button to submit search
  userEvent.click(button);
  // verify search box is there
  expect(screen.getByLabelText('search-box')).toBeInTheDocument();
  // verify search worked submission worked and new page has loaded
  screen.getByText('Results for dogs');
});
