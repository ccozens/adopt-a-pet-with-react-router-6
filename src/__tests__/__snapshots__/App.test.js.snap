// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should look like the snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <div
        class="nav-logo"
      >
        <img
          alt="Petlover"
          src="logo.svg"
        />
        <form
          class="search-form"
        >
          <input
            aria-label="search-box"
            class="search"
            type="text"
          />
          <button
            class="search-button"
            type="submit"
          >
            ðŸ”Ž
          </button>
        </form>
      </div>
      <ul
        class="nav-links"
      >
        <li>
          <a
            aria-current="page"
            class="nav-link-active"
            href="/"
          >
            All Pets
          </a>
        </li>
      </ul>
    </nav>
    <h2>
      Loading...
    </h2>
  </div>
</DocumentFragment>
`;
